tests/infra/ccf.py

```diff
@@ -462,22 +462,27 @@ def find_primary_and_any_backup(self, timeout=3):
     def wait_for_all_nodes_to_catch_up(self, primary, timeout=3):
         """
         Wait for all nodes to have joined the network and globally replicated
-        all transactions executed on the primary (including the transactions
+        all transactions globally executed on the primary (including transactions
         which added the nodes).
         """
-        with primary.node_client() as c:
-            res = c.get("getCommit")
-            local_commit_leader = res.commit
-            term_leader = res.term
+        for _ in range(timeout):
```

Comment:
We've swapped out these `for range(timeout)` loops with `while`s so that the timeouts specify time rather than iterations (if `getCommit` takes 5 seconds, this loop may take ~18 seconds, which seems unrelated to the timeout of 3). Can we do the same here? The total timeout should probably be shared with the loop below (ie - the `end_time` is calculated once at the start of the function and used by both loops)